resources/gcp/environment.go

```diff
@@ -77,12 +77,13 @@ func logIn(ctx *pulumi.Context) {
 		}
 	}
 
-	// Environment variable provided in the CI, to activate the service-account authentication
-	if os.Getenv("GOOGLE_CREDENTIALS_FILE") != "" {
-		fmt.Println("GOOGLE_CREDENTIALS_FILE environment detected, activating service account authentication")
-		cmd := exec.Command("gcloud", "auth", "activate-service-account", "--key-file", os.Getenv("GOOGLE_CREDENTIALS_FILE"))
-		if err := cmd.Run(); err != nil {
-			ctx.Log.Error(fmt.Sprintf("Error running `gcloud auth activate-service-account --key-file $GOOGLE_CREDENTIALS_FILE`: %v", err), nil)
+	// Passing the environment variable is not enough for the gcloud CLI to work, per https://cloud.google.com/docs/authentication/provide-credentials-adc
+	// We need the CLI to be working for accessing GKE cluster with Pulumi Kubernetes provider.
+	if path, ok := os.LookupEnv("GOOGLE_APPLICATION_CREDENTIALS"); ok {
+		cmd := exec.Command("gcloud", "auth", "activate-service-account", "--key-file", path)
```

Comment:
```suggestion
	// TODO add "path" to imports
	if googleCredentialsPath, ok := os.LookupEnv("GOOGLE_APPLICATION_CREDENTIALS"); ok {
		googleCredentialsPath = path.Clean(googleCredentialsPath)
		cmd := exec.Command("gcloud", "auth", "activate-service-account", "--key-file", googleCredentialsPath)
```

Reply:
I'd say this is not really required because `exec.Command` should prevent you from injecting weird things.